### DISCUSSION TITLE
problèmes avec dernière version

### DISCUSSION THREAD

**Discussion by franck70150** [arkypita/LaserGRBL#1287] [Q&A]
LaserGrbl v4.0.5.0
Grbl v1.1l
Wrapper: UsbSerial2
Win 10, Professional, 10.0.19042.0 (7)
CLR: C:\Windows\Microsoft.NET\Framework64\v4.0.30319\mscorlib.dll 4.8.4300.0 built by: NET48REL1LAST_C

TypeOf exception  [System.DivideByZeroException]
Exception message [Tentative de division par zéro.]
Exception source  [mscorlib], thread [Main Thread]
Exception method  [Void FCallDivide(System.Decimal ByRef, System.Decimal ByRef)]

   ----------- stack trace -----------
   à System.Decimal.FCallDivide(Decimal& d1, Decimal& d2)
   à LaserGRBL.RasterConverter.ConvertSizeAndOptionForm.RefreshPerc() dans E:\LaserGRBL\LaserGRBL\RasterConverter\ConvertSizeAndOptionForm.cs:ligne 186
   à LaserGRBL.UserControls.NumericInput.IntegerInputBase.SetCurrentValue(Int32 value, Boolean byuser) dans E:\LaserGRBL\LaserGRBL\UserControls\NumericInput\IntegerInputBase.cs:ligne 191
   à LaserGRBL.UserControls.NumericInput.IntegerInputRanged.set_CurrentValue(Int32 value) dans E:\LaserGRBL\LaserGRBL\UserControls\NumericInput\IntegerInputRanged.cs:ligne 135
   à LaserGRBL.RasterConverter.ConvertSizeAndOptionForm.ShowDialog(Form parent, ImageProcessor processor) dans E:\LaserGRBL\LaserGRBL\RasterConverter\ConvertSizeAndOptionForm.cs:ligne 131
   à LaserGRBL.RasterConverter.RasterToLaserForm.BtnCreateClick(Object sender, EventArgs e) dans E:\LaserGRBL\LaserGRBL\RasterConverter\RasterToLaserForm.cs:ligne 197
   à System.Windows.Forms.Control.OnClick(EventArgs e)
   à System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   à System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   à System.Windows.Forms.Control.WndProc(Message& m)
   à System.Windows.Forms.ButtonBase.WndProc(Message& m)
   à System.Windows.Forms.Button.WndProc(Message& m)
   à System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by arkypita** [comment]
Hi, and thanks for the bug report. 
As it is a software issue, the correct place is the ["issue" section](https://github.com/arkypita/LaserGRBL/issues).

It seems evident that the error arises from an unexpected division by zero. The data in question is the SMax value which represents 100% of the power. LaserGRBL does not do a test to verify that it is non-zero, and this causes it to crash.

I will improve LaserGRBL to intercept and handle this unexpected zero value, but you should understand why on your machine there is this zero, because it shouldn't be.

I am talking about the value $30. This parameter can be 255, 1000, or any other arbitrary value that will be used to indicate 100% of the power, but NOT zero. What number do you have at $30? Can you confirm it is zero on your engraver? Do you know why?

![image](https://user-images.githubusercontent.com/8782035/104950508-13c26800-59c1-11eb-8e1d-8401254c54b8.png)

**Comment by franck70150** [comment]
bonjour, merci pour la réponse . c'était bien en valeur zero en $ 30  j'ai corrigé est ça marche en mettant valeur a 255  merci encore pour la réponse et beau travail pour le logiciel .
